src/components/NodeLink/functionUpdateVis.js

```diff
@@ -242,8 +236,15 @@ function updateVis() {
     .attr("rx", () => nodeMarkerLength / 2)
     .attr("ry", () => nodeMarkerHeight / 2);
 
+  let nodeClasses = colorVariable === "table" ? [...new Set(graphStructure.nodes.map(value => value[colorVariable]))] : 
+    [...new Set(graphStructure.nodes.map(value => value["id"].split("/")[0]))];
+
+  let nodeColorScale = d3.scaleOrdinal()
+    .domain(nodeClasses)
+    .range(nodeColors);
```

Comment:
This is fine--it gives you very fine control over the colors and how they are mapped--but it would be simpler to do something like

```javascript
let nodeColorScale = d3.scaleOrdinal(d3.schemeCategory10)
```

When you create a D3 color scale like this, you can just throw values at it and get colors back, and they will be consistent based on what order individual values are invoked. Then you wouldn't have to recompute the list of classes as you do here. And since this color scale could be used for the lifetime of the component, you could even just make it a member of the components data props.

This is mainly just for information. The code is fine as you have it, but could be simplified (while yielding that fine grained control over the colors) to something like what I suggested.

Reply:
So you don't have to specify a specific domain? It just maps inputs (consistently) to the same values?

Reply:
Yeah, it'll map each new input it encounters to the next color (or something along those lines), and then each old input will remain mapped to whatever color came back the first time.

Reply:
Okay. I'll move it over to the vue component props so it doesn't recompute each time

Reply:
Fixed in a286cff